design/crd-status.md

```diff
@@ -0,0 +1,316 @@
+# Exposing the status with CRDs
+
+## Summary
+
+The purpose of this enhancement is to expose useful information to perform troubleshooting
+via CRDs.
+
+## Motivation
+
+Exposing part of the internal state was not possible until the introduction of CRD
+based configuration, and despite part of the status is exposed via Prometheus
+metrics, troubleshooting MetalLB often requires inspecting the logs of the various
+controllers, and it's not always easy to understand why a service is not working.
+
+For example, inspecting logs is required when:
+
+* an invalid configuration was applied
+* the session with a BGP peer is not established
+* a service was not announced to a BGP peer
+
+### Goals
+
+* Expose useful information to ease the debug
+
+A high level list of informations we want to retrieve is:
+
+* configuration status (valid/invalid and reason)
+* number of used / available IPs per IPAddressPool
+* session state for both BGP and BFD, per node - peer
+* service announcement status, per peer - node (BGP)
+* service announcement node (L2)
+
+### Non-Goals
+
+* Providing BGP status for the legacy implementation
+* Providing a Status subsection for each resource
+
+## Proposal
+
+### User Stories
```

Comment:
wdyt about detailing a bit here or in a next section how the user would achieve these with the proposal?
I think it would add some value here and form some sort of (non-guaranteed) base for the future docs